RestaurantReservationSystem.API/Controllers/RestaurantsController.cs

```diff
@@ -0,0 +1,224 @@
+        /// <param name="patchDoc">The JSON Patch document containing the operations.</param>
+        /// <returns>The updated restaurant or appropriate error messages.</returns>
+        [HttpPatch("{id}")]
+        public async Task<IActionResult> PatchAsync(int id, [FromBody] JsonPatchDocument<RestaurantRequest> patchDoc)
+        {
+            if (patchDoc == null)
+                return BadRequest(ApiResponse<string>.FailResponse("Patch document cannot be null"));
+
+            var existingRestaurant = await _restaurantService.GetByIdAsync(id);
+            if (existingRestaurant == null)
+                return NotFound(ApiResponse<string>.FailResponse("Restaurant not found"));
+
+            var restaurantToPatch = new RestaurantRequest
+            {
+                Name = existingRestaurant.Name,
+                Address = existingRestaurant.Address,
+                PhoneNumber = existingRestaurant.PhoneNumber,
+                OpeningHours = existingRestaurant.OpeningHours
+            };
+
+            patchDoc.ApplyTo(restaurantToPatch, ModelState);
+
+            if (!ModelState.IsValid)
+                return BadRequest(ApiResponse<string>.FailResponse("Invalid patch document"));
+
+            if (!TryValidateModel(restaurantToPatch))
+                return BadRequest(ModelState);
+
+            var updatedRestaurant = await _restaurantService.UpdateAsync(id, restaurantToPatch);
+            if (updatedRestaurant == null)
+                return NotFound(ApiResponse<string>.FailResponse("Failed to update restaurant"));
+
+            return Ok(ApiResponse<RestaurantResponse>.SuccessResponse(updatedRestaurant));
+        }
+
+        /// <summary>
+        /// Deletes a restaurant by ID.
+        /// </summary>
+        /// <param name="id">The ID of the restaurant to delete.</param>
+        /// <returns>Success message or 404 if not found.</returns>
+        [HttpDelete("{id}")]
+        public async Task<IActionResult> DeleteAsync(int id)
+        {
+            var deletedRestaurant = await _restaurantService.DeleteAsync(id);
+            if (!deletedRestaurant)
+                return NotFound(ApiResponse<string>.FailResponse("Restaurant not found"));
+
+            return Ok(ApiResponse<string>.SuccessResponse("Restaurant deleted successfully"));
+        }
+
+        /// <summary>
+        /// Retrieves all employees working at the specified restaurant.
+        /// </summary>
+        /// <param name="id">The unique identifier of the restaurant.</param>
+        /// <returns>
+        /// An <see cref="IActionResult"/> containing a successful <see cref="ApiResponse{T}"/> with a list
+        /// of <see cref="EmployeeResponse"/> if the restaurant exists; otherwise, a not found response.
+        /// </returns>
+        [HttpGet("{id}/employees")]
+        public async Task<IActionResult> GetEmployeesAsync(int id)
+        {
+            var restaurant = await _restaurantService.GetByIdAsync(id);
```

Comment:
all these checks should be moved to the services level `GetEmployeesByRestaurantIdAsync`, and throw from there a not found exception. then you can have a middleware to map these exceptions to status codes. same comment for the other controllers.